### DISCUSSION TITLE
Conditional M2O fields / filters in the app

### DISCUSSION THREAD

**Discussion by H14** [directus/directus#11844] [Archive]
Hi

I can't seem to get conditional fields or filters to work (or they do not work as I expected).

**I have a collection called 'Test'.** It has the following fields:
<img width="836" alt="image" src="https://user-images.githubusercontent.com/91126835/155518681-6f2ca57b-44de-4f40-b36d-3e61fe9b8481.png">

Graad, Domein and Studierichting are very basic M2O fields. They relate to the following collections:

* graden
   * id
   * omschrijving (text)

* domeinen
  * id
   * omschrijving (text)

* studierichtingen
   * id
   * omschrijving (text)
   * domein (M2O > Domeinen)
 
---
 
I'm trying to accomplish the following.
- **Studierichtingen should be hidden, and should only be shown when the selected 'graad' from 'graden' = id 4.**
  - I made the studierichten **Hidden on Detail** and added the following condition: <img width="824" alt="image" src="https://user-images.githubusercontent.com/91126835/155519784-3337b393-7bc2-46f6-9370-35700d9e113c.png"> This does not work, it stays hidden no matter what.

- **The conditional studierichtingen-field should be filtered based on the selected 'domein' from 'domeinen'.** 
  - I don't know how to do this, filters seem to only accept hard coded values and not references to other field values? <img width="402" alt="image" src="https://user-images.githubusercontent.com/91126835/155520441-1476d510-90e5-4dd7-88f3-374ff9c25aad.png">

Could someone please point me in the right direction?

Many thanks

Hans

**Comment by Jeyakumarrathnaselvi** [comment]
It is needed for M2M and M2M as well.  Can someone help us implement this

**Comment by rijkvanzanten** [comment]
Heya, this is a confusing in what data is being saved to the API. When you select an existing item, the value of the field becomes `field = 1`, not `field = { id: 1 }`. Therefore, when you configure the validation to check against `{ id: 1 }` it won't pass the validation. In this case, if you only want to check against selecting the existing item, set the validation to be `domein equals 4`.

Also see https://github.com/directus/directus/issues/7423#issuecomment-902293310, https://github.com/directus/directus/discussions/7123#discussioncomment-1121173, https://github.com/directus/directus/issues/7423#issuecomment-910918474

**Comment by H14** [comment]
I understand, but as a non-technical user, this is the default behavior when I use the GUI made for non-technical users. This is very unintuitive and misleading. The rules interface is well polished, but you can only get it to work by fiddling with the raw values because of technical reasons.

Also: the GUI adds numbers as a string in the JSON, like this "id": "1". This doesn't work either in this context, it only works without quotes in the raw value "id": 1.

**Reply by rijkvanzanten** [reply]
Yup! I'm aware, hence my suggestion in https://github.com/directus/directus/issues/7423#issuecomment-910918474 🙂

**Comment by SergeyRe** [comment]
i made extension in order to solve m2m and m2o conditions confuse
sorry -its hardcoded -mainly for idea demonstration 
[
directus-extension-m2m-conditions](https://github.com/SergeyRe/directus-extension-m2m-conditions)
